ccm_web/client/src/api.ts

```diff
@@ -1,36 +1,63 @@
+import { CanvasCourseBase } from './models/canvas'
 import { Globals, HelloMessageData } from './models/models'
 import handleErrors from './utils/handleErrors'
 
 export interface LtiProps {
   ltiKey: string | undefined
 }
 
-const createAuthHeaders = (key: string | undefined): RequestInit => {
+const initRequest = (key: string | undefined, headers: string[][] = []): RequestInit => {
   if (key !== undefined) {
+    headers.push(['Authorization', 'Bearer ' + key])
     return {
       credentials: 'include',
-      headers: { Authorization: 'Bearer ' + key }
+      headers: headers
     }
   }
   return {}
 }
 
+const getGet = (key: string | undefined): RequestInit => {
+  const request = initRequest(key)
+  request.method = 'GET'
+  return request
+}
+
+// This currently assumes all put requests have a json payload and recieve a json response
+const getPut = (key: string | undefined, body: string): RequestInit => {
+  const headers: string[][] = []
+  headers.push(['Content-Type', 'application/json'])
+  headers.push(['Accept', 'application/json'])
+  const request = initRequest(key, headers)
+  request.method = 'PUT'
+  request.body = body
+  return request
+}
+
+export const getCourse = async (key: string | undefined, courseId: number): Promise<CanvasCourseBase> => {
+  const request = getGet(key)
+  const resp = await fetch(`/api/course/${courseId}/name`, request)
```

Comment:
What should be the UI response when we fail getting the course? It is currently displaying as error getting the course name. Should we leave it blank?

Reply:
I'd think about rendering an error alert (some red box or other layout element) in the place of the title with a message about what happened.

Reply:
Hm, forgot about the snackbar, which may make such an alert repetitive. Some placeholder ("Unknown") might be okay.

Reply:
I'm not a UX person but I feel like using the snackbar should be more for interactive things, rather than for an error happening on page load.. does that sound reasonable?  Just seems like something I have not come across in the wild.

Reply:
Ah, yeah, that seems valid. So you think think the in-place alert for automatic processes is okay?

Reply:
>  does that sound reasonable?

Agreed

With that said we can discuss this with Janel and melinda. 

My 2 cents would be leave it blank for instructor user, and Admin user we can default it some name Course name ( who can edit). 

We can chat with Janel and Melinda once they come back